content/compute-app/non-functional-requirements/readme.md

```diff
@@ -0,0 +1,156 @@
+   - Some other requirements are hidden - for example, if you're an ISV, you might not know until it's too late that one of your customers assumes they will have their own database, and when they discover that you share the database with other      tenants it causes a major problem.
+   - And some requirements are made up by the business, and don't really have any direct consequence. For example, you might aim to have your APIs respond within 500ms. If you miss the mark, nobody is going to sue you, but the business cares about this because it's part of their quality and performance goal.
+   - Distinguish SLAs (service level agreements) vs. SLOs (service level objectives).
+  - For each requirement, it's helpful to have a clear understanding of whether the requirement is an SLA or an SLO, who "owns" the requirement, who is responsible for monitoring compliance with the requirement, who is responsible for rectifying any violations, and what happens if it's violated. A lot of the time these aren't written down or understood systematically, but they should be.
+
+- **Tradeoffs**. An important part of solution architecture is deciding on tradeoffs. For example, imagine you're building an enterprise solution. Your security team wants you to lock down your environment, but your development team is telling you that will cause them major problems and make everything take much longer, which means your timeframes will be at risk. What should you do?
+
+- **WAF pillars.** Talk about the fact that the five WAF pillars are there to help to structure our thinking and conversations about the key areas of NFRs.
+
+  The rest of this session focuses on each of the pillars in turn, and talks about the key NFRs in each, using them both as a way to explain the requirements and also as a way to think systematically about non-functional requirements overall.
+
+## Reliability
+
+- **Pillar Concept** - HA/BC/DR - scenarios, availability and recoverability
+
+- **Uptime** is the amount of time that your solution is available and responsive, usually measured as a percentage of time (e.g. 99.9%).
+  - In general, the higher the uptime, the more resiliency you need, and often the more expensive the solution is to run.
+  - We calculate composite SLAs to tell you what Azure is guaranteeing for your solution, although there's more to it than just a single number.
+
+- Recovery from various types of problems is another key element of resiliency.
+- Imagine you've got a solution that you decide to spread across multiple regions in a primary/secondary or hot/cold architecture.
+- Now, a disaster hits your primary region. How long does it take to recover?
+   - In a hot-cold architecture, you might need to stand up some more infrastructure and wait for it to be deployed and configured.
+   - In a warm-standby architecture, you might need to trigger a failover and then wait for DNS records to be updated.
+   - This is your recovery time.
```

Comment:
```suggestion
   - This is your recovery time objective.
```

Reply:
For this specific instance I did mean "recovery time" - as in, "for this example, this is the time it takes to recover".

I then thought we'd say "Is this acceptable? If not, then you need to set your recovery time _objective_ to be smaller than that." or similar.

Does that make sense?

Reply:
Yes, makes perfect sense!